homeassistant/components/netatmo/camera.py

```diff
@@ -189,128 +162,107 @@ def handle_event(self, event: dict) -> None:
-        self._vpnurl, self._localurl = self._data.camera_urls(self._id)
-        self._status = camera.get("status")
-        self._sd_status = camera.get("sd_status")
-        self._alim_status = camera.get("alim_status")
-        self._is_local = camera.get("is_local")
-        self._attr_is_streaming = bool(self._status == "on")
-
-        if self._model == "NACamera":  # Smart Indoor Camera
-            self.hass.data[DOMAIN][DATA_EVENTS][self._id] = self.process_events(
-                self._data.events.get(self._id, {})
-            )
-        elif self._model == "NOC":  # Smart Outdoor Camera
-            self.hass.data[DOMAIN][DATA_EVENTS][self._id] = self.process_events(
-                self._data.outdoor_events.get(self._id, {})
-            )
+        self._attr_is_on = self._camera.alim_status is not None
+        self._attr_available = self._camera.alim_status is not None
+
+        if self._camera.monitoring is not None:
+            self._attr_is_streaming = self._camera.monitoring
+            self._attr_motion_detection_enabled = self._camera.monitoring
+
+        self.hass.data[DOMAIN][DATA_EVENTS][self._id] = self.process_events(
+            self._camera.events
+        )
 
         self._attr_extra_state_attributes.update(
             {
                 "id": self._id,
-                "status": self._status,
-                "sd_status": self._sd_status,
-                "alim_status": self._alim_status,
-                "is_local": self._is_local,
-                "vpn_url": self._vpnurl,
-                "local_url": self._localurl,
+                "monitoring": self._monitoring,
+                "sd_status": self._camera.sd_status,
+                "alim_status": self._camera.alim_status,
+                "is_local": self._camera.is_local,
+                "vpn_url": self._camera.vpn_url,
+                "local_url": self._camera.local_url,
                 "light_state": self._light_state,
             }
         )
 
-    def process_events(self, events: dict) -> dict:
+    def process_events(self, event_list: list[NaEvent]) -> dict:
         """Add meta data to events."""
-        for event in events.values():
-            if "video_id" not in event:
+        events = {}
+        for event in event_list:
+            if not (video_id := getattr(event, "video_id")):
                 continue
-            if self._is_local:
-                event[
-                    "media_url"
-                ] = f"{self._localurl}/vod/{event['video_id']}/files/{self._quality}/index.m3u8"
-            else:
-                event[
-                    "media_url"
-                ] = f"{self._vpnurl}/vod/{event['video_id']}/files/{self._quality}/index.m3u8"
+            event_data = event.__dict__
+            event_data["subevents"] = [
+                event.__dict__
+                for event in event_data.get("subevents", [])
+                if not isinstance(event, dict)
+            ]
```

Comment:
This looks like you've made some poor decisions in the library, what do you think? Can't this data be provided in a better way?

Reply:
Chances are that this can and will be removed completely. I am just waiting for feedback from Netatmo devs.

Reply:
Just looked at this again, adding this comment just for the sake of it, it should probably be a dict comprehension instead and not reusing the event object in both layers of the statement :)